src/app/reporting/Engine.cpp

```diff
@@ -100,15 +108,38 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeDataList(ReportData::Builder &
     {
         if (clusterInfo->IsDirty())
         {
-            AttributeDataElement::Builder attributeDataElementBuilder = attributeDataList.CreateAttributeDataElementBuilder();
             if (apReadHandler->IsInitialReport())
             {
                 // Retrieve data for this cluster instance and clear its dirty flag.
-                err = RetrieveClusterData(attributeDataElementBuilder, *clusterInfo);
+                err = RetrieveClusterData(attributeDataList, *clusterInfo);
                 VerifyOrExit(err == CHIP_NO_ERROR,
                              ChipLogError(DataManagement, "<RE:Run> Error retrieving data from cluster, aborting"));
                 attributeClean = false;
             }
+            else
+            {
+                for (auto & path : mDirtyPathList)
+                {
+                    if (path.IsDirty())
```

Comment:
why is this check here? The member names is 'mDirtyPathlist' so I imagine the content being 'dirty paths'. Is this  untrue and should we change the member name? or not do the dirty check?